src/H5D.c

```diff
@@ -279,12 +279,21 @@ H5Dcreate_anon(hid_t loc_id, hid_t type_id, hid_t space_id, hid_t dcpl_id, hid_t
     FUNC_ENTER_API(H5I_INVALID_HID)
     H5TRACE5("i", "iiiii", loc_id, type_id, space_id, dcpl_id, dapl_id);
 
+    /* Sanity check */
+    if (loc_id < 0 || type_id < 0 || space_id < 0 || dcpl_id < 0 || dapl_id < 0)
```

Comment:
I don't think these are necessary. `dcpl_id` and `dapl_id` will be checked by H5P_isa_class below. `loc_id` will be checked by H5I_object. There might be a case for `type_id` and `space_id`, but I'm assuming those will fail a bit further when we try to get an object out of them.

Reply:
Corrected.